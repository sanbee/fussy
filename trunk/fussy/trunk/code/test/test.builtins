test(){
auto x,y,x1,x2
x=1pm0.1
y=2pm0.2
x1=1pm0.1
x2=1pm0.1
print "Value of x=\t",x,"\n"
print "Value of y=\t",y,"\n"
print "Value of x1=\t",x1,"\n"
print "Value of x2=\t",x2,"\n"
print "sin(x)=\t",sin(x),"\n"
print "sqrt(1-sin(x)^2)=\t",sqrt(1-sin(x)^2),"\n"
print "cos(x)=\t",cos(x),"\n"
print "tan(x)=\t",tan(x),"\n"
print "sin(x)/cos(x)=\t",sin(x)/cos(x),"\n"

print "asin(sin(x))=\t",asin(sin(x)),"\n"
print "acos(cos(x))=\t",acos(cos(x)),"\n"
print "atan(tan(x))=\t",atan(tan(x)),"\n"

print "asinh(sinh(x))=\t",asinh(sinh(x)),"\n"
print "acosh(cosh(x))=\t",acosh(cosh(x)),"\n"
print "atanh(tanh(x))=\t",atanh(tanh(x)),"\n"

print "exp(ln(x))=\t",exp(ln(x)),"\n";
print "sinh(x)=\t",sinh(x),"\n"
print "(exp(x)-exp(-x))/2=\t",(exp(x)-exp(-x))/2,"\n"
print "sinh(x)/((exp(x)-exp(-x))/2)=\t",sinh(x)/((exp(x)-exp(-x))/2),"\n"
print "acosh(cosh(x))=\t",acosh(cosh(x)),"\n"
print "atanh(tanh(x))=\t",atanh(tanh(x)),"\n"

print "exp(ln(x))=",exp(ln(x)),"\n"
print "x/exp(ln(x))=",x/exp(ln(x)),"\n"

print "y/(1-y) =",(y/(1-y))%16.6f,"\n";
print "y/(-y+1)=",(y/(-y+1))%16.6f,"\n"

print "sin(x1)*sin(x1)=\t",sin(x1)*sin(x1),"\n";
print "sin(x1)*sin(x2)=\t",sin(x1)*sin(x2),"\n";
print "(sin(x1)^2+cos(x1)^2)=\t",(sin(x1)^2+cos(x1)^2),"\n";
print "(sin(x1)^2+cos(x1)^2)/(sin(x2)^2+cos(x2)^2)=\t",(sin(x1)^2+cos(x1)^2)/(sin(x2)^2+cos(x2)^2),"\n";
print "sin(x1)^2+cos(x2)^2=\t",sin(x1)^2+cos(x2)^2,"\n";
};
test();
