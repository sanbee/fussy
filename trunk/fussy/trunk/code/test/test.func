/* -*- C -*- */
/* calling b should give a seg fault */
/* LocalSymbTab needs to be changed to a list instead of a vector */
/* 
   The above is done.  In fact the code would have failed earlier
   if e() was called even without going to lists!
*/
t=0;
fa(x){return (x^2);}
fb(x) {auto t; t:=x;return x;}
fc(x) {return 2;}
fd(x){return sin(fa(x));}
fd1(x){ auto t;t:=sin(fa(x));return t;}
fd2(x){ auto t;t=sin(fa(x));return t;}

x=2pm0.2;
y=x;
NL="\n";
line="-------------------------------------------\n";

f(x)
{
  auto c,e;
  c=30pm1;
  e:=c*x
  return e;
}

test()
{
  print "fb(x) = \t",fb(x),NL;
  print "Following should have same value and error: ",NL,line;
  print "fd(x) = \t",fd(x),NL;
  print "fd1(x) = \t",fd1(x),NL;
  print "sin(fa(x)) = \t",sin(fa(x)),NL;
  print "sin(x^2) = \t",sin(x^2),NL;
  print "fd2(x) = \t",fd2(x),NL;
  print "Following should have same value and error: ",NL,line;
  print "x*fd(x) = \t",x*fd(x),NL;
  print "x*fd1(x) = \t",x*fd1(x),NL;
  print "x*sin(fa(x)) = \t",x*sin(fa(x)),NL;
  print "x*sin(x*x) = \t",x*sin(x*x), NL;
  y=x;
  line="--------------------------------------------------------------\n";
  print "Following should have same value as above but different error: ",NL,line;
  print "y*fd1(x) = \t",y*fd1(x),NL;
  print "y*sin(fa(x)) = \t",y*sin(fa(x)),NL;
  print "y*sin(x^2) = \t", y*sin(x^2),NL;
  print "x*fd2(x) = \t",x*fd2(x),NL;
  print "Value of f(11) = ",f(11),"\n";
};

test();

/*
a(x){return cos(x);}
b(x){auto t;t=1.1;t=a(x);print "from b: ",t,"\n";}
c(x){auto t,t0;t0=10*x; t=t0;print "from c: ",t,"\n";print a(x),t0,"\n";}
d(x){return a(x);}
e(x){auto t,t0;t0=10*x; t=t0;print "from e: ",t,"\n";b(x);print d(x),t0,"\n";}
*/
