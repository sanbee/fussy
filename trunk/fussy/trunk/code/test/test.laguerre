/*
    Recursive generating formula for Laguerre polynomial and it's
    derivative.

    Shows that f(n,x).rms == (df(n,x).val*x.rms) (which is
    f'(n,x)*deltaX) 
*/

f(n,x)
{
  if (n<=0) return 1; 
  if (n==1) return -x+1; 
  return ((2*n-1-x)*f(n-1,x)-(n-1)*f(n-2,x))/n;
}

df(n,x)
{
  return (n*f(n,x)-n*f(n-1,x))/x;
}

test(x)
{
   auto n,i,y;
   for(i=1;i<10;i++) {print "l(",i%-6.2f,",",x%-6.2f,")= ",f(i,x)," ",fabs(df(i,x).val*x.rms),"\n";};
}

test(3pm1);
