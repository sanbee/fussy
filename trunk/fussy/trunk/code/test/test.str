f(x){auto y;y="test = ";print y,x,"\n";}

