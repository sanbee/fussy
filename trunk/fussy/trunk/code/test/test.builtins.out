Value of x=	   1.00000 +/-    0.10000
Value of y=	   2.00000 +/-    0.20000
Value of x1=	   1.00000 +/-    0.10000
Value of x2=	   1.00000 +/-    0.10000
sin(x)=	   0.84147 +/-    0.05403
sqrt(1-sin(x)^2)=	   0.54030 +/-    0.08415
cos(x)=	   0.54030 +/-    0.08415
tan(x)=	   1.55741 +/-    0.34255
sin(x)/cos(x)=	   1.55741 +/-    0.34255
asin(sin(x))=	   1.00000 +/-    0.10000
acos(cos(x))=	   1.00000 +/-    0.10000
atan(tan(x))=	   1.00000 +/-    0.10000
asinh(sinh(x))=	   1.00000 +/-    0.10000
acosh(cosh(x))=	   1.00000 +/-    0.10000
atanh(tanh(x))=	   1.00000 +/-    0.10000
exp(ln(x))=	   1.00000 +/-    0.10000
sinh(x)=	   1.17520 +/-    0.15431
(exp(x)-exp(-x))/2=	   1.17520 +/-    0.15431
sinh(x)/((exp(x)-exp(-x))/2)=	   1.00000
acosh(cosh(x))=	   1.00000 +/-    0.10000
atanh(tanh(x))=	   1.00000 +/-    0.10000
exp(ln(x))=   1.00000 +/-    0.10000
x/exp(ln(x))=   1.00000
y/(1-y) =       -2.000000 +/-         0.200000
y/(-y+1)=       -2.000000 +/-         0.200000
sin(x1)*sin(x1)=	   0.70807 +/-    0.09093
sin(x1)*sin(x2)=	   0.70807 +/-    0.06430
(sin(x1)^2+cos(x1)^2)=	   1.00000 +/-    0.00000
(sin(x1)^2+cos(x1)^2)/(sin(x2)^2+cos(x2)^2)=	   1.00000 +/-    0.00000
sin(x1)^2+cos(x2)^2=	   1.00000 +/-    0.12859
